### PR TITLE
dpdk: 23.07 -> 23.11

### PR DESCRIPTION
**Pull request by wfdewith**
## Description of changes

DPDK 23.11 removes the kernel network interface driver (`kni`), so we no longer need to build it for every kernel. I've also updated `dpdk-kmods`, which includes a kernel module for DPDK, and `pktgen`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wfdewith** [comment]
Result of `nixpkgs-review pr 297787` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>redpanda-server</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_4_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_lqx.dpdk-kmods</li>
  </ul>
</details>
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>dpdk</li>
    <li>dpdk.doc</li>
    <li>linuxKernel.packages.linux_4_19.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_4_19.odp-dpdk</li>
    <li>linuxKernel.packages.linux_4_19.pktgen</li>
    <li>linuxKernel.packages.linux_5_10.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_10_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_15.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_15_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_4.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_1.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_1_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_6.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_hardened.dpdk-kmods (linuxKernel.packages.linux_6_6_hardened.dpdk-kmods)</li>
    <li>linuxKernel.packages.linux_6_7.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_7_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_8.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_latest_libre.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_libre.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_xanmod.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_xanmod_latest.dpdk-kmods (linuxKernel.packages.linux_xanmod_stable.dpdk-kmods)</li>
    <li>linuxKernel.packages.linux_zen.dpdk-kmods</li>
    <li>spdk</li>
  </ul>
</details>

**Comment by wfdewith** [comment]
> <details>
>  <summary>3 packages failed to build:</summary>
>  <ul>
>    <li>linuxKernel.packages.linux_4_19_hardened.dpdk-kmods</li>
>    <li>linuxKernel.packages.linux_5_4_hardened.dpdk-kmods</li>
>    <li>linuxKernel.packages.linux_lqx.dpdk-kmods</li>
>  </ul>
> </details>

These appear to be unrelated to this change and do not build on `master` either.

File: pkgs/os-specific/linux/dpdk-kmods/default.nix [hunk 1]
```diff
@@ -2,20 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "dpdk-kmods";
-  version = "2022-08-29";
+  version = "2023-02-05";
```
Comment (Mic92) [inline pkgs/os-specific/linux/dpdk-kmods/default.nix:5]: Also it's not this issue is not introduced by this PR.
Could you add `broken = kernel.isHardened` to meta? Than we no longer need to test the build for this kernel. Thanks!

**Review by Mic92 — APPROVED** [review]

File: pkgs/os-specific/linux/dpdk-kmods/default.nix (hunk 1 above)
Comment (wfdewith) [reply pkgs/os-specific/linux/dpdk-kmods/default.nix:5]: Done, though this is what happens on master:

```
$ nix-build -A linuxKernel.packages.linux_lqx.dpdk-kmods
[...]
error: builder for '/nix/store/5s5v974a5m9vh2yf4611ls1ha7ama76w-linux-config-6.7.9.drv' failed with exit code 255;
       last 25 log lines:
       > GOT: * Runtime Testing
       > GOT: *
       > QUESTION: Runtime Testing, NAME: RUNTIME_TESTING_MENU, ALTS: N/y/?, ANSWER: n
       > GOT: n
       > QUESTION: Memtest, NAME: MEMTEST, ALTS: Y/n/?, ANSWER: y
       > GOT: y
       > QUESTION: Microsoft Hyper-V driver testing, NAME: HYPERV_TESTING, ALTS: N/y/?, ANSWER:
       > GOT:
       > GOT: *
       > GOT: * Rust hacking
       > GOT: *
       > GOT: #
       > GOT: # configuration written to .config
       > GOT: #
       > GOT: make[1]: Leaving directory '/build/source/build'
       > GOT: make: Leaving directory '/build/source'
       > warning: unused option: CFS_BANDWIDTH
       > warning: unused option: MLX5_CORE_EN
       > warning: unused option: NUMA_BALANCING
       > warning: unused option: PSI
       > warning: unused option: RT_GROUP_SCHED
       > warning: unused option: SCHED_AUTOGROUP
       > warning: unused option: SCHED_CORE
       > error: unused option: UCLAMP_TASK
       > error: unused option: UCLAMP_TASK_GROUP
       For full logs, run 'nix log /nix/store/5s5v974a5m9vh2yf4611ls1ha7ama76w-linux-config-6.7.9.drv'.
error: 1 dependencies of derivation '/nix/store/mrkklqr7jaddazda0wl1gcy8i35hwqyi-linux-6.7.9.drv' failed to build
error: 1 dependencies of derivation '/nix/store/h1rdrcmr2ysjacf3nd2cvcxh9l388r92-dpdk-kmods-2022-08-29.drv' failed to build
```

which may be related to #296867, but I haven't investigated further.

Comment (wfdewith) [reply pkgs/os-specific/linux/dpdk-kmods/default.nix:5]: Also note that it does build for some other hardened kernels

**Comment by Kamillaova** [comment]
Maybe update spdk and other packages that are heavily related to dpdk?

**Comment by Kamillaova** [comment]
Result of `nixpkgs-review pr 297787` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_10_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_15_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_4_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_1_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_6_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_7_hardened.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_hardened.dpdk-kmods</li>
    <li>linuxPackages_4_19_hardened.dpdk-kmods</li>
    <li>linuxPackages_5_10_hardened.dpdk-kmods</li>
    <li>linuxPackages_5_15_hardened.dpdk-kmods</li>
    <li>linuxPackages_5_4_hardened.dpdk-kmods</li>
    <li>linuxPackages_6_1_hardened.dpdk-kmods</li>
    <li>linuxPackages_6_6_hardened.dpdk-kmods</li>
    <li>linuxPackages_6_7_hardened.dpdk-kmods</li>
    <li>linuxPackages_hardened.dpdk-kmods</li>
    <li>redpanda-server</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>linuxPackages_lqx.dpdk-kmods (linuxKernel.packages.linux_lqx.dpdk-kmods)</li>
  </ul>
</details>
<details>
  <summary>18 packages built:</summary>
  <ul>
    <li>dpdk (linuxKernel.packages.linux_4_19.dpdk ,linuxKernel.packages.linux_4_19_hardened.dpdk ,linuxKernel.packages.linux_5_10.dpdk ,linuxKernel.packages.linux_5_10_hardened.dpdk ,linuxKernel.packages.linux_5_15.dpdk ,linuxKernel.packages.linux_5_15_hardened.dpdk ,linuxKernel.packages.linux_5_4.dpdk ,linuxKernel.packages.linux_5_4_hardened.dpdk ,linuxKernel.packages.linux_6_1.dpdk ,linuxKernel.packages.linux_6_1_hardened.dpdk ,linuxKernel.packages.linux_6_6.dpdk ,linuxKernel.packages.linux_6_6_hardened.dpdk ,linuxKernel.packages.linux_6_7.dpdk ,linuxKernel.packages.linux_6_7_hardened.dpdk ,linuxKernel.packages.linux_6_8.dpdk ,linuxKernel.packages.linux_hardened.dpdk ,linuxKernel.packages.linux_latest_libre.dpdk ,linuxKernel.packages.linux_libre.dpdk ,linuxKernel.packages.linux_lqx.dpdk ,linuxKernel.packages.linux_xanmod.dpdk ,linuxKernel.packages.linux_xanmod_latest.dpdk ,linuxKernel.packages.linux_xanmod_stable.dpdk ,linuxKernel.packages.linux_zen.dpdk ,linuxPackages-libre.dpdk ,linuxPackages.dpdk ,linuxPackages_4_19_hardened.dpdk ,linuxPackages_5_10_hardened.dpdk ,linuxPackages_5_15_hardened.dpdk ,linuxPackages_5_4_hardened.dpdk ,linuxPackages_6_1_hardened.dpdk ,linuxPackages_6_6_hardened.dpdk ,linuxPackages_6_7_hardened.dpdk ,linuxPackages_hardened.dpdk ,linuxPackages_latest-libre.dpdk ,linuxPackages_latest.dpdk ,linuxPackages_lqx.dpdk ,linuxPackages_xanmod.dpdk ,linuxPackages_xanmod_latest.dpdk ,linuxPackages_xanmod_stable.dpdk ,linuxPackages_zen.dpdk)</li>
    <li>dpdk.doc (linuxKernel.packages.linux_4_19.dpdk.doc ,linuxKernel.packages.linux_4_19_hardened.dpdk.doc ,linuxKernel.packages.linux_5_10.dpdk.doc ,linuxKernel.packages.linux_5_10_hardened.dpdk.doc ,linuxKernel.packages.linux_5_15.dpdk.doc ,linuxKernel.packages.linux_5_15_hardened.dpdk.doc ,linuxKernel.packages.linux_5_4.dpdk.doc ,linuxKernel.packages.linux_5_4_hardened.dpdk.doc ,linuxKernel.packages.linux_6_1.dpdk.doc ,linuxKernel.packages.linux_6_1_hardened.dpdk.doc ,linuxKernel.packages.linux_6_6.dpdk.doc ,linuxKernel.packages.linux_6_6_hardened.dpdk.doc ,linuxKernel.packages.linux_6_7.dpdk.doc ,linuxKernel.packages.linux_6_7_hardened.dpdk.doc ,linuxKernel.packages.linux_6_8.dpdk.doc ,linuxKernel.packages.linux_hardened.dpdk.doc ,linuxKernel.packages.linux_latest_libre.dpdk.doc ,linuxKernel.packages.linux_libre.dpdk.doc ,linuxKernel.packages.linux_lqx.dpdk.doc ,linuxKernel.packages.linux_xanmod.dpdk.doc ,linuxKernel.packages.linux_xanmod_latest.dpdk.doc ,linuxKernel.packages.linux_xanmod_stable.dpdk.doc ,linuxKernel.packages.linux_zen.dpdk.doc ,linuxPackages-libre.dpdk.doc ,linuxPackages.dpdk.doc ,linuxPackages_4_19_hardened.dpdk.doc ,linuxPackages_5_10_hardened.dpdk.doc ,linuxPackages_5_15_hardened.dpdk.doc ,linuxPackages_5_4_hardened.dpdk.doc ,linuxPackages_6_1_hardened.dpdk.doc ,linuxPackages_6_6_hardened.dpdk.doc ,linuxPackages_6_7_hardened.dpdk.doc ,linuxPackages_hardened.dpdk.doc ,linuxPackages_latest-libre.dpdk.doc ,linuxPackages_latest.dpdk.doc ,linuxPackages_lqx.dpdk.doc ,linuxPackages_xanmod.dpdk.doc ,linuxPackages_xanmod_latest.dpdk.doc ,linuxPackages_xanmod_stable.dpdk.doc ,linuxPackages_zen.dpdk.doc)</li>
    <li>linuxKernel.packages.linux_4_19.dpdk-kmods</li>
    <li>odp-dpdk (linuxPackages.odp-dpdk)</li>
    <li>pktgen (linuxPackages.pktgen)</li>
    <li>linuxKernel.packages.linux_5_10.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_15.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_5_4.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_6_1.dpdk-kmods</li>
    <li>linuxPackages.dpdk-kmods (linuxKernel.packages.linux_6_6.dpdk-kmods)</li>
    <li>linuxKernel.packages.linux_6_7.dpdk-kmods</li>
    <li>linuxPackages_latest.dpdk-kmods (linuxKernel.packages.linux_6_8.dpdk-kmods)</li>
    <li>linuxPackages_latest-libre.dpdk-kmods (linuxKernel.packages.linux_latest_libre.dpdk-kmods)</li>
    <li>linuxPackages-libre.dpdk-kmods (linuxKernel.packages.linux_libre.dpdk-kmods)</li>
    <li>linuxPackages_xanmod.dpdk-kmods (linuxKernel.packages.linux_xanmod.dpdk-kmods)</li>
    <li>linuxPackages_xanmod_latest.dpdk-kmods (linuxKernel.packages.linux_xanmod_latest.dpdk-kmods ,linuxPackages_xanmod_stable.dpdk-kmods)</li>
    <li>linuxPackages_zen.dpdk-kmods (linuxKernel.packages.linux_zen.dpdk-kmods)</li>
    <li>spdk</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Can we mark it broken for LQX so we don't spend time building it for nothing?

**Review by abuibrahim — APPROVED** [review]

**Comment by Mic92** [comment]
> Can we mark it broken for LQX so we don't spend time building it for nothing?

There is currently no way to figure out if the current kernel is an LQX kernel. So we would need a new flag there. Since this PR only touches DPDK and doesn't introduce new regressions in this area, I would consider this out of scope for this change.

**Pull request merged**